app/importers/dropbox.py

```diff
@@ -0,0 +1,183 @@
+            crud.job.update_status(
+                self.db,
+                db_obj=job,
+                status=JobStatus.IMPORTING_ERROR_AUTHORIZATION_REQUIRED,
+            )
+            return
+
+        url_details = get_url_details(job.import_url)
+
+        assert url_details
+
+        self.url_type = url_details["type"]
+
+        try:
+            self.build_tree_recursively(self.tree_root["children"], url_details["path"])
+            self.download_tree_recursively(self.tree_root["children"], job.path)
+        except (AuthError, HttpError, BadInputError):
+            crud.job.update_status(
+                self.db,
+                db_obj=job,
+                status=JobStatus.IMPORTING_ERROR_AUTHORIZATION_REQUIRED,
+            )
+            return
+        except (ApiError):
+            crud.job.update_status(
+                self.db,
+                db_obj=job,
+                status=JobStatus.IMPORTING_ERROR_DATA_UNREACHABLE,
+            )
+            return
+
+        manifest_input = ManifestInput(job_id=job.id, source_url=job.import_url)
+
+        root_entry = self.tree_root.get("entry")
+        assert root_entry
+
+        self.root_entry = root_entry
+        manifest_input.project_name = root_entry.name
+
+        # Set the number of total_items
+        crud.job.update_total_items(
+            self.db, job_id=self.job_id, total_items=job.imported_items
+        )
+
+        crud.manifest.update_or_create(self.db, obj_in=manifest_input)
+
+        crud.job.update_status(
+            self.db, db_obj=job, status=JobStatus.IMPORTING_SUCCESSFULLY
+        )
+
+    def get_entries_from_folder(self, folder_path: str) -> List[dropbox.files.Metadata]:
+
+        entries = []
+
+        if self.url_type == "shared":
+
+            # Change the type to id to treat the following folders as ids
+            self.url_type = "id"
+
+            link = dropbox.files.SharedLink(
+                url=folder_path,
+            )
+
+            # When handling a shared_link folder, treat the first iteration different
+            result = self.dropbox_handler.files_list_folder(path="", shared_link=link)
+
+        elif self.url_type == "user_folder" or self.url_type == "id":
+            result = self.dropbox_handler.files_list_folder(folder_path)
+
+        entries.extend(result.entries)
+
+        while result.has_more:
+            result = self.dropbox_handler.files_list_folder_continue(result.cursor)
+            entries.extend(result.entries)
+
+        return entries
+
+    def build_tree_recursively(self, current_level: Dict, folder_url: str) -> None:
+
+        entries = self.get_entries_from_folder(folder_url)
+
+        for entry in entries:
+            name = entry.name
+            current_level[name] = {"entry": entry, "children": {}}
+
+        for node in current_level.values():
+            entry = node.get("entry")
+            if isinstance(entry, dropbox.files.FileMetadata):
```

Comment:
this could be removed

Reply:
I would leave it for the next step, which would be the hooks for the files

Reply:
Ok, it's fine I guess if  you plan to extend it immediately after